src/buildblock/interpolate_projdata.cxx

```diff
@@ -200,96 +205,313 @@ interpolate_projdata(ProjData& proj_data_out,
+      for (index_out[1] = min_out[1]; index_out[1] <= max_out[1]; ++index_out[1])
+        {
+          for (index_out[2] = min_out[2]; index_out[2] <= max_out[2]; ++index_out[2])
+            {
+              for (index_out[3] = min_out[3]; index_out[3] <= max_out[3]; ++index_out[3])
+                {
+                  // translate index on output to coordinate
+                  auto bin
+                = Bin(0 /* segment */, index_out[2] /* view */, 
+                index_out[1] /* axial pos */, index_out[3] /* tangential pos
+                */);
+                  auto out_m = proj_data_out_info.get_m(bin);
+                  double axial_idx = (out_m - m_offset) / m_sampling;
+                  int axial_floor = std::floor(axial_idx);
+                  int axial_ceil = std::ceil(axial_idx);
+                  if (axial_floor == axial_ceil)
+                    {
+                      if (axial_floor == 0)
+                        axial_ceil++;
+                      else
+                        axial_floor--;
+                    }
+
+                  // find the two crystals for this bin and on which buckets (modules) they are
+                  int det1_num_out, det2_num_out;
+                  const auto proj_data_out_info_ptr = dynamic_cast<const ProjDataInfoGenericNoArcCorr*>(&proj_data_out_info);
+                  proj_data_out_info_ptr->get_det_num_pair_for_view_tangential_pos_num(det1_num_out,
+                                                                                       det2_num_out,
+                                                                                       index_out[2], /*view*/
+                                                                                       index_out[3] /* tangential pos */);
+                  const int dets_per_module_out = proj_data_out_info.get_scanner_sptr()->get_num_transaxial_crystals_per_bucket();
+                  const int det1_module = std::floor(det1_num_out / dets_per_module_out);
+                  const int det2_module = std::floor(det2_num_out / dets_per_module_out);
+
+                  // translate the crystal position on each module from the full size scanner to the downsampled scanner
+                  const auto proj_data_in_info_ptr = dynamic_cast<const ProjDataInfoGenericNoArcCorr*>(&proj_data_in_info);
+                  const int dets_per_module_in
+                      = proj_data_in_info_ptr->get_scanner_sptr()->get_num_transaxial_crystals_per_bucket();
+                  double crystal1_out_module_idx = det1_num_out % dets_per_module_out;
+                  double crystal1_out_module_pos
+                      = static_cast<double>(
+                            std::floor(crystal1_out_module_idx
+                                       / proj_data_out_info_ptr->get_scanner_sptr()->get_num_transaxial_crystals_per_block()))
+                            * proj_data_out_info_ptr->get_scanner_sptr()->get_transaxial_block_spacing()
```

Comment:
`std::floor` again not needed as `int` division. If that's not what you want, do the `double cast on one of the elements.

Reply:
in the first line the static_cast<double> is actually not needed, so I will remove it.